### PR TITLE
More user friendly conversions

### PR DESCRIPTION
**Pull request by Flix6x**
Mostly reindexing from belief_time to belief_horizon, but also getting rid of future warning and one bug fix where selecting the most recent belief can result in selecting a NaN belief, which isn't the one we want to select.

I still have 1 test (test_crps) to fix.

### CODE REVIEW

File: timely_beliefs/beliefs/utils.py [hunk 1]
```diff
@@ -445,10 +445,13 @@ def set_reference(
     if return_expected_value is True:
         reference_df = reference_df.for_each_belief(get_expected_belief)
 
+    belief_timing_col = (
+        "belief_time" if "belief_time" in reference_df.index.names else "belief_horizon"
```
Comment (nhoening) [inline timely_beliefs/beliefs/utils.py:459]: Not sure I understand exactly what is happening here. But "belief_horizon" is used as a name in the concatenated and returned dataframe (lines 458ff). Is that still accurate when we dropped it?

**Review by nhoening — APPROVED** [review]

File: timely_beliefs/beliefs/utils.py (hunk 1 above)
Comment (Flix6x) [reply timely_beliefs/beliefs/utils.py:457]: It is. We first get rid of the belief timing column and then reconstruct our own, where we choose the horizon perspective explicitly.

**Comment by Flix6x** [comment]
[Get rid of future_warning](https://trello.com/c/aS9Yt3ef/66-get-rid-of-futurewarning)

**Comment by Flix6x** [comment]
[Convert belief timing index level in more places](https://trello.com/c/kqYsSvGF/64-convert-belief-timing-index-level-in-more-places)

**Pull request merged**